app/controllers/application_controller.rb

```diff
@@ -6,16 +6,63 @@ def log_in(user)
   end
 
   def log_out
+    forget(current_user)
     session.delete(:user_id)
     @current_user=nil
   end
 
+  def forget(user)
+    user.forget
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end 
+
+
+  helper_method :current_user
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user && user.authenticated?(:remember, cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end 
+    end 
+  end
+
+  helper_method :logged_in? 
+  def logged_in?
+    !current_user.nil?
```

Comment:
I usually avoid double negatives as they're harder to read. How about `current_user.present?`. In rails, there's usually opposites on most of the convenience methods.